### DISCUSSION TITLE
skynet_context命名问题

### DISCUSSION THREAD

**Discussion by TonyWoo** [cloudwu/skynet#1915] [Q&A]
分析大部分代码并且也阅读了相关的wiki，skynet设计确实很巧妙和高效。这里有个疑问，[skynet_context](https://github.com/cloudwu/skynet/blob/master/skynet-src/skynet_server.c#L42)应该整个系统核心对象之一。从使用上来看它代表着类似于OS下的进程或线程的一个结构体。本质上应该就是代表了一个actor。

但这里好奇的是为什么要命名context这样中文里面相当于是“参数”或“上下文“这样非常通用，但其实并无具体用意的名称。从可读性来讲，感觉不是很好理解。读起来像”skynet上下文“对象...

是不是有更具体的，更能代表其用途和身份的名称？比如skynet_actor_context?

**Comment by cloudwu** [comment]
我不记得，但 context 是一个多线程程序中常见术语，就好比面向对象中的 object 。

比如 opengl 的每个实例就叫 context 。
